### DISCUSSION TITLE
I would like to render a simple table and be it as terse as possible

### DISCUSSION THREAD

**Discussion by robstoll** [ajalt/mordant#196] [Q&A] [answered]
Following an example:
```
table {
    body {
        this.align = TextAlign.LEFT
        this.cellBorders = Borders.ALL
        row {
            cell("a longer text spanning two columns") { columnSpan = 2 }
            cells("A", "B")
        }
        row {
            cell("short")
            cells("some text", "C", "D")  }
    }
}
```
![image](https://github.com/user-attachments/assets/2811e6ce-01a3-46fe-827e-7c9626c95873)

As you can see, there is an extra space after `columns`. 
Also, I would like that the cell which contains `short` does not take up more space than necessary. How can I achieve this?

**Comment by ajalt** [comment]
Set the [padding][1] to 0 

```
┌───────────────────────────────────┬─┬─┐
│a longer text spanning two columns │A│B│
├─────────────────┬─────────────────┼─┼─┤
│short            │some text        │C│D│
└─────────────────┴─────────────────┴─┴─┘
```

[1]: https://ajalt.github.io/mordant/api/mordant/com.github.ajalt.mordant.table/-cell-style-builder/padding.html

**Reply by robstoll** [reply]
Alright, that's already better, thanks 🙂 and how to set that the cell containing "short" takes only as much space as necessary?

**Reply by ajalt** [reply, marked as answer]
You'll have to use fixed column widths for the first two columns if you want to do that.

If that long span cell wasn't there, you could use a `Custom` width with a low priority to do what you want, but if a cell spans columns and is wider than the content of those columns, all the spanned `Auto` columns will have equal width.

**Reply by robstoll** [reply]
Thanks for the quick reply, will see if that fits my need
